packages/desktop/src/providers/network/variables.rs

```diff
@@ -1,18 +1,214 @@
-use serde::Serialize;
+use netdev::interface::InterfaceType as NdInterfaceType;
+use netdev::ip::{Ipv4Net as NdIpv4Net, Ipv6Net as NdIpv6Net};
+use netdev::mac::MacAddr as NdMacAddr;
+use serde::{Serialize, Serializer};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
 #[derive(Serialize, Debug, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct NetworkVariables {
+  pub default_interface: NetworkInterface,
+  pub default_gateway: Gateway,
   pub interfaces: Vec<NetworkInterface>,
 }
 
 #[derive(Serialize, Debug, Clone)]
-#[serde(rename_all = "camelCase")]
 pub struct NetworkInterface {
   pub name: String,
-  pub mac_address: String,
-  pub transmitted: u64,
-  pub total_transmitted: u64,
-  pub received: u64,
-  pub total_received: u64,
+  pub friendly_name: Option<String>,
+  pub description: Option<String>,
+  #[serde(serialize_with = "interfacetype_ser")]
+  pub if_type: NdInterfaceType,
+  #[serde(serialize_with = "ipv4_ser")]
+  pub ipv4: Vec<NdIpv4Net>,
+  #[serde(serialize_with = "ipv6_ser")]
+  pub ipv6: Vec<NdIpv6Net>,
+  #[serde(serialize_with = "macaddr_ser")]
+  pub mac_addr: NdMacAddr,
+  pub transmit_speed: Option<u64>,
+  pub receive_speed: Option<u64>,
+  pub dns_servers: Vec<IpAddr>,
+  pub default: bool,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct Gateway {
+  #[serde(serialize_with = "macaddr_ser")]
+  pub mac_addr: NdMacAddr,
+  pub ipv4: Vec<Ipv4Addr>,
+  pub ipv6: Vec<Ipv6Addr>,
+  pub ssid: String,
+  pub signal_strength_percent: u32,
+  pub connected: bool,
```

Comment:
could we rename:
`mac_addr` -> `mac_address`
`signal_strength_percent` -> `signal_strength`
`connected` -> `is_connected`

don't know enough about networking to make a judgment on this, but would it make sense to rename `ipv4` and `ipv6` to `ipv4_address` and `ipv6_address`?

Reply:
actions, and ipv4/6 -> addresses to represent array of addresses